lib/logger/lib/logger.ex

```diff
@@ -939,8 +939,14 @@ defmodule Logger do
   def __should_log__(level, module) do
     level = elixir_level_to_erlang_level(level)
 
-    if :logger_config.allow(level, module) do
-      level
+    case get_process_level(self()) do
+      nil ->
+        if :logger_config.allow(level, module) do
+          level
+        end
+
+      process_level ->
```

Comment:
We still need to check if this level is allowed by either calling `:logger_config.allow(level)` or by calling `:logger_config.allow(level, module)` if we want still want to look up the module.

Reply:
Calling :logger_config.allow(level, module) there would give precedence for module/primary level. Pushed some fixes and I believe we have the expected behaviour now :)